.github/workflows/publish.yml

```diff
@@ -1,84 +1,81 @@
 name: Publish CLI
 
 on:
-  release:
-    types:
-      - published
+  push:
     branches:
-      - master
-    tags:
-      - v*.*.*
+      - main
+      - rc
+      - arc-*
 
 env:
   CI: true
   NPM_CACHE_NAME: cli-node-modules
 
 jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Get npm cache directory
+        id: npm-cache
+        run: |
+          echo "::set-output name=dir::$(npm config get cache)"
+      - uses: actions/cache@v2
+        with:
+          path: ${{ steps.npm-cache.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - run: npm ci
+      - run: npm run lint
+      - run: npm run check-circular
+  test:
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
```

Comment:
Wanted to note that node 12 EOL is the end of this month, and node 17 is out. We might want to take this chart into consideration

https://nodejs.org/en/about/releases/

Reply:
We should do this as another PR but good callout.

Reply:
Sure, made a ticket for that